source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -909,6 +913,29 @@ See the `Load Balancer Specification <../load-balancers/load-balancers.rst#event
        *  The ID of the Connection
        */
       connectionId: int64;
+
+      /**
+       * The time it took to establish the connection.
+       * In accordance with the definition of establishment of a connection
+       * specified by `ConnectionPoolOptions.maxConnecting`,
+       * it is the time elapsed between the `ConnectionCreatedEvent`
+       * emitted by the same checking out and this event.
+       *
+       * Naturally, this duration is not greater than
+       * `ConnectionCheckedOutEvent`/`ConnectionCheckOutFailedEvent.duration`.
+       *
+       * A driver that delivers events synchronously MUST NOT include in this duration
+       * the time to deliver the `ConnectionCreatedEvent`.
+       * Doing so eliminates a thing to worry about in support cases related to this duration,
+       * because the time to deliver synchronously is affected by user code.
+       * The driver MUST document this behavior
+       * as well as explicitly warn users that the behavior may change in the future.
```

Comment:
can you elaborate on this?  I'm not too familiar with how emitting events synchronously would be impacted by user's code.

Reply:
Synchronous delivery with necessity means that checking out waits for some sort of a confirmation from user code that an emitted event was delivered. If the user code delays the confirmation for whatever reason, it will affect the duration reported by the driver despite the driver having nothing to do with the delay. In the most simplest form it may be invocation of a synchronous user function and passing an emitted event as an argument to it, as part of checking out.

Reply:
Ah gotcha, a callback mechanism or something similar as opposed to event emitters or streams.  Thanks!